lib/jnpr/junos/exception.py

```diff
@@ -297,3 +298,26 @@ class ConnectClosedError(ConnectError):
     def __init__(self, dev):
         ConnectError.__init__(self, dev=dev)
         dev.connected = False
+
+
+class JSONLoadError(Exception):
+
+    """
+    Generated if json content of rpc reply fails to load
+    """
+    def __init__(self, exception, rpc_content):
+        self.ex_msg = str(exception)
+        self.rpc_content = rpc_content
+        self.offending_line = ''
+        obj = re.search('line (\d+)', self.ex_msg)
+        if obj:
+            line_no = int(obj.group(1))
+            self.offending_line = rpc_content.splitlines()[line_no-1]
+
+    def __repr__(self):
+        return "{0}(reason: {1}, \nThe offending config appears to be: \n{2}" \
+                .format(self.__class__.__name__, self.ex_msg,
+                        self.offending_line)
```

Comment:
I like your idea of providing 5 lines (2 pre + 1 + 2 post) instead of just the single offending line. In addition, I suggest you print these 5 lines with line numbers prepended to the actual config line.